core/gui/src/app/workspace/component/code-editor-dialog/code-editor.component.ts

```diff
@@ -421,51 +504,77 @@ export class CodeEditorComponent implements AfterViewInit, SafeStyle, OnDestroy
-          const languageClient = new MonacoLanguageClient({
-            name: "Python UDF Language Client",
-            clientOptions: {
-              documentSelector: ["python"],
-              errorHandler: {
-                error: () => ({ action: ErrorAction.Continue }),
-                closed: () => ({ action: CloseAction.Restart }),
-              },
-            },
-            connectionProvider: { get: () => Promise.resolve({ reader, writer }) },
-          });
-          languageClient.start();
-          reader.onClose(() => languageClient.stop());
-        }
-      };
+  private initDiffEditor(): void {
+    if (this.wrapper) {
+      from(this.wrapper.dispose(true))
+        .pipe(takeUntil(this.componentDestroy))
+        .subscribe({
+          next: () => this.initializeDiffEditor(),
+        });
+    } else {
+      this.initializeDiffEditor();
     }
   }
 
-  private initDiffEditor() {
-    if (this.code) {
-      this.editor = monaco.editor.createDiffEditor(this.editorElement.nativeElement, {
-        readOnly: true,
-        theme: "vs-dark",
-        fontSize: 11,
-        automaticLayout: true,
-      });
+  private initializeDiffEditor(): void {
+    if (this.code && !this.wrapper) {
+      this.wrapper = new MonacoEditorLanguageClientWrapper();
+      const fileSuffix = this.getFileSuffixByLanguage(this.language);
       const currentWorkflowVersionCode = this.workflowActionService
         .getTempWorkflow()
         ?.content.operators?.filter(
           operator =>
             operator.operatorID ===
             this.workflowActionService.getJointGraphWrapper().getCurrentHighlightedOperatorIDs()[0]
         )?.[0].operatorProperties.code;
-      this.editor.setModel({
-        original: monaco.editor.createModel(this.code.toString(), "python"),
-        modified: monaco.editor.createModel(currentWorkflowVersionCode, "python"),
-      });
+
+      const userConfig: UserConfig = {
+        wrapperConfig: {
+          editorAppConfig: {
+            $type: "extended",
+            codeResources: {
+              main: {
+                text: currentWorkflowVersionCode,
+                uri: `in-memory-${this.operatorID}-version.${fileSuffix}`,
+              },
+              original: {
+                text: this.code.toString(),
+                uri: `in-memory-${this.operatorID}.${fileSuffix}`,
+              },
+            },
+            useDiffEditor: true,
+            diffEditorOptions: {
+              readOnly: true,
+            },
+            userConfiguration: {
+              json: JSON.stringify({
+                "workbench.colorTheme": "Default Dark Modern",
+              }),
+            },
+          },
+        },
+      };
+      if (this.language === "python") {
+        userConfig.languageClientConfig = {
+          languageId: "python",
+          options: {
+            $type: "WebSocketUrl",
+            url: "ws://localhost:3000/language-server",
```

Comment:
Use `getWebsocketUrl("/python-language-server", "3000")`, check the original code.